src/Compiler/Driver/CompilerDiagnostics.fs

```diff
@@ -1337,18 +1337,19 @@ type Exception with
                     |> List.exists (fun prods ->
                         let prodIds =
                             prods
-                            |> List.map Parser.prodIdxToNonTerminal
-                            |> List.map (fun nonTerminal ->
-                                match nonTerminal with
-                                | NONTERM_Category_Type -> Parser.NONTERM_typ
-                                | NONTERM_Category_Expr -> Parser.NONTERM_declExpr
-                                | NONTERM_Category_Pattern -> Parser.NONTERM_atomicPattern
-                                | NONTERM_Category_IfThenElse -> Parser.NONTERM_ifExprThen
-                                | NONTERM_Category_SignatureFile -> Parser.NONTERM_signatureFile
-                                | NONTERM_Category_ImplementationFile -> Parser.NONTERM_implementationFile
-                                | NONTERM_Category_Definition -> Parser.NONTERM_moduleDefn
-                                | NONTERM_Category_Interaction -> Parser.NONTERM_interaction
-                                | nt -> nt)
+                            |> List.map (
+                                Parser.prodIdxToNonTerminal
+                                >> function
```

Comment:
We don't use function in new compiler code, match is easier to debug.

Reply:
Makes sense. Maybe create an issue to easier debugging function? For example: Why doesn't the "function" tooltip show the captured variable value?

Reply:
Reverted.